autogluon/utils/tabular/ml/models/catboost/catboost_model.py

```diff
@@ -64,6 +75,9 @@ def _fit(self, X_train, Y_train, X_test=None, Y_test=None, time_limit=None, **kw
                 num_classes = self.num_classes
             else:
                 num_classes = 10  # Guess if not given, can do better by looking at y_train
+        elif self.problem_type == SOFTCLASS:
+            num_classes = Y_train.shape[1]
+            self.num_classes = num_classes
```

Comment:
I added this for safety.  The code above it reads:
```
if self.num_classes is not None:
                num_classes = self.num_classes
            else:
                num_classes = 10  # Guess if not given, can do better by looking at y_train
```
which leads me to believe it may sometimes not be set?

Reply:
Added TODO that we can delete this if unnecessary, but don't think it will introduce errors.